src/models/ProjectDocumentRepository.ts

```diff
@@ -0,0 +1,99 @@
+import { promiseMap } from "../migrations/utils";
+import { D2Api } from "../types/d2-api";
+import { getUid } from "../utils/dhis2";
+import { Id, ProjectDocument } from "./ProjectDocument";
+
+export class ProjectDocumentRepository {
+    private api: D2Api;
+    constructor(api: D2Api) {
+        this.api = api;
+    }
+
+    async getByIds(ids: Id[]): Promise<ProjectDocument[]> {
+        const d2Response = await this.api.models.documents
+            .get({
+                fields: {
+                    id: true,
+                    url: true,
+                    name: true,
+                },
+                filter: {
+                    id: {
+                        in: ids,
+                    },
+                },
+            })
+            .getData();
+
+        const documents = d2Response.objects.map(d2Document => {
+            return ProjectDocument.create({
+                blob: undefined,
+                id: d2Document.id,
+                name: d2Document.name,
+                url: d2Document.url,
+                href: `${this.api.apiPath}/documents/${d2Document.id}/data`,
+                sizeInBytes: 0,
+                sharing: undefined,
+            });
+        });
+
+        const projectDocuments = await promiseMap(documents, async projectDocument => {
+            const file = await this.api
+                .request<D2FileResource>({
+                    method: "get",
+                    url: `/fileResources/${projectDocument.url}`,
+                })
+                .getData();
+
+            return ProjectDocument.create({
+                ...projectDocument,
+                blob: undefined,
+                sizeInBytes: Number(file.contentLength),
+            });
+        });
+        return projectDocuments;
+    }
+
+    async saveAll(documents: ProjectDocument[]): Promise<ProjectDocument[]> {
```

Comment:
Question: when attached files are removed, the files themselves are kept in the system, right? is that a problem?

Reply:
I've added the code to remove documents. File resources are going to be deleted automatically by dhis.

https://docs.dhis2.org/en/full/develop/dhis-core-version-236/developer-manual.html#webapi_file_resources_constraints